apps/aestratum/src/aestratum.erl

```diff
@@ -0,0 +1,447 @@
+    S1 = case payout_tx_persisted(TxHash, TopHeight) of
+             true ->
+                 ?TXN(case length(aestratum_db:payments(Height)) == 1 of
+                          true  -> aestratum_db:delete_records(Height);
+                          false -> aestratum_db:delete_payment(TxId)
+                      end),
+                 S#aestratum_state{pending_tx_id = undefined,
+                                   pending_tx_hash = undefined};
+             false ->
+                 S
+         end,
+    {noreply, S1};
+handle_info(chain_payment_tx_check, S) ->
+    {noreply, S};
+
+handle_info(candidate_cache_cleanup, S) ->
+    ?TXN(aestratum_db:delete_candidates_older_than(
+           aestratum_conv:delta_secs_to_universal_datetime(
+             -trunc(?CANDIDATE_CACHE_CLEANUP_INTERVAL / 1000)))),
+    {noreply, S};
+
+handle_info({gproc_ps_event, stratum_new_candidate,
+             #{info := [{HeaderBin, #candidate{block = BlockCand} = C, Target, _} = _Info]}},
+            #aestratum_state{} = State) ->
+    case aec_blocks:is_key_block(BlockCand) of
+        true ->
+            KH = aec_blocks:to_header(BlockCand),
+            BlockHash  = aestratum_miner:hash_data(HeaderBin),
+            {ok, _}    = ?TXN(aestratum_db:store_candidate(BlockHash, HeaderBin, C)),
+            TargetInt  = aeminer_pow:scientific_to_integer(Target),
+            ChainEvent = #{event => recv_block,
+                           block => #{block_target  => TargetInt,
+                                      block_hash    => aestratum_conv:hex_encode(BlockHash),
+                                      block_version => aec_headers:version(KH)}},
+            ?INFO("new candidate with target ~p", [TargetInt]),
+            aestratum_user_register:notify({chain, ChainEvent});
+        false ->
+            ok
+    end,
+    {noreply, State};
+
+handle_info(Req, State) ->
+    ?ERROR("spurious info request ~p", [Req]),
+    {noreply, State}.
+
+
+handle_cast({submit_share, Miner, MinerTarget, CandidateBlockHash}, State) ->
+    ?TXN(aestratum_db:store_share(Miner, MinerTarget, CandidateBlockHash)),
+    {noreply, State};
+
+handle_cast({submit_solution, CandidateBlockHash, Nonce, Pow}, State) ->
+    case ?TXN(aestratum_db:get_candidate(CandidateBlockHash)) of
+        {ok, #aestratum_candidate{header = HeaderBin,
+                                  record = #candidate{block = CandidateBlock}}} ->
+            KeyBlock        = aec_blocks:set_nonce_and_pow(CandidateBlock, Nonce, Pow),
+            KeyBlockHeader  = aec_blocks:to_header(KeyBlock),
+            {ok, BlockHash} = aec_headers:hash_header(KeyBlockHeader),
+            ok = ?TXN(aestratum_db:mark_share_as_solution(CandidateBlockHash, BlockHash)),
+            ?INFO("got solution for blockhash ~p (nonce = ~p)", [BlockHash, Nonce]),
+            aec_conductor ! {stratum_reply, {{ok, {Nonce, Pow}}, HeaderBin}};
```

Comment:
What prevents on node start a submit request from a client to trigger this code path with the `aec_conductor` process not registered yet?

It seems the dependencies in the init shall be clarified.

Reply:
I've added dependency on aecore, thanks for suggestion.